src/typechecker.erl

```diff
@@ -5112,7 +5112,7 @@ record_field_types(Fields) ->
               end, Fields).
 
 % Helper returning the type of module_info/0 and module_info/1 functions
--spec get_module_info_type(arity()) -> fun_ty().
+-spec get_module_info_type(arity()) -> type() | [type()].
```

Comment:
Can we use an two-clause spec (0) -> type(); (1) -> [type()] here?

Reply:
Btw, what happens if a code contains a call to `module_info/N` where N > 1? Do we crash?